crypto/rand/drbg_lib.c

```diff
@@ -285,21 +397,56 @@ int RAND_DRBG_generate(RAND_DRBG *drbg, unsigned char *out, size_t outlen,
 }
 
 /*
- * Set the callbacks for entropy and nonce.  We currently don't use
- * the nonce; that's mainly for the KATs
+ * Set the RAND_DRBG callbacks for obtaining entropy and nonce.
+ *
+ * In the following, the signature and the semantics of the
+ * get_entropy() and cleanup_entropy() callbacks are explained.
+ *
+ * GET_ENTROPY
+ *
+ *     size_t get_entropy(RAND_DRBG *ctx,
+ *                        unsigned char **pout,
+ *                        int entropy,
+ *                        size_t min_len, size_t max_len);
+ *
+ * This is a request to allocate and fill a buffer of size
+ * |min_len| <= size <= |max_len| (in bytes) which contains
+ * at least |entropy| bits of entropy. The buffers address is
+ * to be returned in |*pout| and the number of collected
+ * randomness bytes as return value.
+ *
+ * If the callback succeeds to acquire |entropy| bits of entropy,
+ * it returns the number of bytes written to the buffer
+ * (which may be less than the allocated size of the buffer).
```

Comment:
There is a little bit of duplication here with the previous paragraph in terms of returning the number of collected/written randomness bytes, but I don't really mind it.